python/understack-workflows/understack_workflows/helpers.py

```diff
@@ -1,30 +1,35 @@
 import argparse
 import logging
 import os
-import sushy
 import sys
 
+import sushy
+
 logger = logging.getLogger(__name__)
 
 
-def setup_logger(name):
-    logger = logging.getLogger(name)
-    handler = logging.StreamHandler()
-    formatter = logging.Formatter(
-        "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
+def setup_logger(name: str | None = None, level: int = logging.DEBUG):
+    """Standardize our logging.
+
+    Configures the root logger to prefix messages with a timestamp
+    and to output the log level we want to see by default.
+
+    params:
+    name: logger hierarchy or root logger
+    level: default log level (DEBUG)
+    """
+    logging.basicConfig(
+        format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+        level=logging.DEBUG,
     )
-    handler.setFormatter(formatter)
-    logger.addHandler(handler)
-    logger.setLevel(logging.DEBUG)
-    return logger
+    return logging.getLogger(name)
```

Comment:
doesn't this change the format and level for root and all future loggers instead of affecting a single instance?
If so, is there a risk that loggers initialized later by the workflow code will spam too much?

Reply:
Yes. But the result with an installed package is that all the helper modules will be parallel to the binary and won't produce any logging. Is that what we want out of the box?

Reply:
I was thinking that we may want to enable debug just for the code we own, not everything down the line. Let's try your version, we can always revert this later.